rpc/ExperimentalConsumer.ts

```diff
@@ -0,0 +1,245 @@
+    if (this.blockPendings.length) {
+      const controller = new AbortController()
+      // TODO: why not emitting?
+      this.subscription("chainHead_unstable_body", undefined, [followId, blockHash], (_result) => {
+        while (this.blockPendings.length) {
+          const blockPending = this.blockPendings.shift()!
+          blockPending.resolve(null!)
+        }
+      }, controller.signal)
+    }
+  }
+
+  extrinsicsPendings: Deferred<Uint8Array>[] = []
+  extrinsics(blockHash?: string) {
+    if (blockHash) return this.archiveConsumer.block(blockHash)
+    const pending = deferred<Uint8Array>()
+    this.extrinsicsPendings.push(pending)
+    return pending
+  }
+  extrinsicsFlush(followId: string, blockHash: string) {
+    if (this.extrinsicsPendings.length) {
+      const controller = new AbortController()
+      this.subscription<{ event: "done"; result: string }>(
+        "chainHead_unstable_body",
+        undefined,
+        [followId, blockHash],
+        (result) => {
+          while (this.extrinsicsPendings.length) {
+            const blockPending = this.extrinsicsPendings.shift()!
+            blockPending.resolve(hex.decode(result.result))
+          }
+        },
+        controller.signal,
+      )
+    }
+  }
+
+  keys(key: Uint8Array, limit: number, start?: Uint8Array, blockHash?: string) {
+    return this.archiveConsumer.keys(key, limit, start, blockHash)
+  }
+
+  valuesKeys: Record<string, true> = {}
+  valuesPendings: [keys: string[], pending: Deferred<Uint8Array[]>][] = []
+  values(keys: Uint8Array[], blockHash?: string) {
+    if (blockHash) return this.archiveConsumer.values(keys, blockHash)
+    const keysEncoded = keys.map((key) => {
+      const encoded = hex.encodePrefixed(key)
+      this.valuesKeys[encoded] = true
+      return encoded
+    })
+    const pending = deferred<Uint8Array[]>()
+    this.valuesPendings.push([keysEncoded, pending])
+    return pending
+  }
+  valuesFlush(followId: string, blockHash: string) {
+    const keys = Object.entries(this.valuesKeys)
+    if (keys.length) {
+      const items = keys.map((key) => ({ key, type: "value" }))
+      const controller = new AbortController()
+      this.subscription<{ event: "items"; items: { key: string; value: string }[] }>(
+        "chainHead_unstable_storage",
+        "chainHead_unstable_stopStorage",
+        [followId, blockHash, items, null],
+        (message) => {
+          const lookup = Object.fromEntries(message.items.map(({ key, value }) => [key, value]))
+          while (this.valuesPendings.length) {
+            const valuesPending = this.valuesPendings.shift()!
+            const [keys, pending] = valuesPending
+            pending.resolve(keys.map((key) => hex.decode(lookup[key]!)))
+          }
+          this.valuesKeys = {}
+          controller.abort()
+        },
+        controller.signal,
+      )
+    }
+  }
+
+  nonce(ss58Address: string) {
+    return this.archiveConsumer.nonce(ss58Address)
+  }
```

Comment:
I think we will need to get this using `chainHead_storage` and query for `system.account(AccountId32)`

<img width="731" alt="image" src="https://github.com/paritytech/capi/assets/1209171/3c588a81-7dd7-458a-8604-0253d5a817c1">

Reply:
There a note in the docs for `chainHead_unstable_call`

> Note: This can be used as a replacement for the legacy state_getMetadata, system_accountNextIndex, and payment_queryInfo functions.

https://paritytech.github.io/json-rpc-interface-spec/api/chainHead_unstable_call.html